### DISCUSSION TITLE
Vercel CLI "No existing credentials found" While deploying inside Vercel.

### DISCUSSION THREAD

**Discussion by ImJustChew** [vercel/vercel#7838] [Help]
I'm having an issue deploying with vercel just after I tried adding vercel as a devDependency.
I had no issues with deploying before. The build step would write
```
Running "vercel build"
10:23:09.584 | Vercel CLI 24.2.1 build (beta) — https://vercel.com/feedback
```

But after installing vercel as a devDependency by `npm install vercel --save-dev`
```
Running "vercel build"
Vercel CLI 24.2.4
Error! No existing credentials found. Please run `vercel login` or pass "--token"
```

Then I tried reversing it by removing vercel from devDependencies but that didn't work either. Now I'm stuck as it can never deploy without a token.
Is there something I'm doing wrong here?

**Comment by cassus** [comment]
As a workaround I found that `--force` avoids this problem.

**Reply by damienlequere** [reply]
Thanks !
It's a workaround but it's very helpfull
